src/gui/base/Icon.ts

```diff
@@ -27,6 +28,13 @@ import("./icons/Icons.js").then((IconsModule) => {
 })
 
 export class Icon implements Component<IconAttrs> {
+	private root?: HTMLElement
```

Comment:
a nitpick but we rather do `root: HTMLElement | null = null`

Reply:
there are probably valid cases where we want optional fields (like in component attributes) so it's a bit hard